src/solvers/td3.ml

```diff
@@ -297,10 +308,26 @@ module WP =
         delete_marked wpoint;
         delete_marked stable;
 
-        print_data data "Data after clean-up"
+
+        print_data data "Data after clean-up";
+
+        List.iter set_start st;
+
+        if not !any_changed_start_state && GobConfig.get_bool "incremental.reluctant" then (
+          (* solve on the return node of changed functions. Only destabilize influenced nodes outside the function if the analysis result changed *)
```

Comment:
This is not necessarily true though, the following `solve` might call `destabilize` something outside this function itself already and still unnecessarily destabilize large parts of the program. In order for this to actually be true, the `solve` from here should (in its all recursive usages) use a different `destabilize`, which comes back and does it reluctantly again or somehow prevent it from going wildly beyond this function.

As it turned out in #365, it's not clear how to even do that properly.

Reply:
You are right, this incremental reluctant destabilization can lead to destabilizing nodes outside of a changed function during the phase of solving it separately. This should be correct even if it does not reduce the number of evaluations as effectively as one could hope. It also ensures that new side-effects are evaluated. I changed the comment because it was not formulated precisely. I think one should look further into how to make the destabilization reluctantly also inside the `solve`. Maybe we should address this in another issue?

Reply:
Yes, this is definitely a separate issue, and at some point we should try to understand what exactly goes wrong when trying to interleave the solve and destabilization. @sim642 When your regressions failed: was it only lack of precision or was it even unsound? This isn't urgent, but would be good to understand if want some intuition about whether it is at all possible to limit destabilization with side-effects.

Reply:
Just imprecise: https://github.com/goblint/analyzer/pull/365/checks?check_run_id=3744468024#step:6:674.